src/dnaio/_core.pyx

```diff
@@ -258,122 +266,127 @@ def fastq_iter(file, sequence_class, Py_ssize_t buffer_size):
                 bufstart += 1
                 bufend += 1
                 extra_newline = True
-            else:
-                break
+            elif last_read_position > record_start:  # Incomplete fastq records are present.
+                    if extra_newline:
```

Comment:
Yes it can. In case of an incomplete record it will first visit the above if branch. Then it will run trough the parser code, only to find that adding the extra linefeed did not help. It will then trigger this if branch. This is necessary because this will report the incomplete record without the extra newline which was added by dnaio, but was not actually there in the input.
If I remove this check, multiple tests will fail with an off-by-one error because of this. You put the check there for a good reason.

I will add a comment to elucidate, given that the reason for the check is not entirely clear.

Reply:
Yes, you’re right. I got confused by the original `break` and somehow assumed that all branches exit the loop, but that’s not the case.